STM32F405RGTx/Core/Src/rtos_threads/ltc6813_thread.cpp

```diff
@@ -0,0 +1,29 @@
+#include "ltc6813_thread.hpp"
+#include "ltc6813.h"
+#include "bms_entry.hpp"
+#include "threads.hpp"
+#include "main.h"
+
+RTOSThread Ltc6813Thread::thread;
+
+void Ltc6813Thread::initialize() {
+    thread = RTOSThread(
+        "ltc6813_thread",
+        1024*2,
+        osPriorityBelowNormal,
+        runDriver
+    );
+}
+
+void Ltc6813Thread::runDriver(void* args) {
+    while (1) {
+        Ltc6813_wakeup_sleep(&ltc6813);
+        Ltc6813_wakeup_idle(&ltc6813);
+        if (Ltc6813_read_adc(&ltc6813, NORMAL_ADC)) {
+            for (uint8_t i = 0; i < NUM_CELLS; i++) {
+                global_bms_data.battery.cells[i].voltage = ltc6813.cell_voltages[i];
+            }
+        }
+        osDelay(LTC_MEASUREMENT_PERIODICITY*1E3);
+    }
+}
```

Comment:
Ok this is kind of a pet peeve of mine but can we get into the habit of adding newlines to the end of our source files? It literally does nothing, but its the convention I guess lmao.